packages/repository/src/decorators/relation.decorator.ts

```diff
@@ -53,10 +65,33 @@ export function relation(definition?: Object) {
  * @param definition
  * @returns {(target:any, key:string)}
  */
-export function belongsTo(definition?: Object) {
-  // Apply model definition to the model class
-  const rel = Object.assign({type: RelationType.belongsTo}, definition);
-  return PropertyDecoratorFactory.createDecorator(RELATIONS_KEY, rel);
+export function belongsTo<T extends typeof Entity>(
+  targetModel: TypeResolver<T>,
```

Comment:
So we don't allow `@belongsTo(User)` any more? Does it have to be `@belongsTo(() => User)`?

Reply:
Yes, this would be the approach we'd need to take if we want to minimize cases where we need to differentiate between `User` and `() => User`